sound/soc/codecs/pcm5102a.c

```diff
@@ -21,22 +21,75 @@
 
 #include <sound/soc.h>
 
+struct pcm5102a_priv {
+	bool rates_384k;
+};
+
+static int pcm5102a_dai_startup(struct snd_pcm_substream *substream,
+	struct snd_soc_dai *dai)
+{
+	struct snd_soc_codec *codec = dai->codec;
+	struct pcm5102a_priv *priv = snd_soc_codec_get_drvdata(codec);
+	int ret;
+
+	if (!priv->rates_384k) {
+		dev_info(codec->dev,
+			 "%s: Limiting sample rate support to 192kHz MAX\n",
+			 __func__);
+
+		dev_dbg(codec->dev, "%s: set minmax (8k/192k) constraint\n",
+			__func__);
+
+		ret = snd_pcm_hw_constraint_minmax(substream->runtime,
+						   SNDRV_PCM_HW_PARAM_RATE,
+						   8000, 192000);
+		if (ret < 0) {
+			dev_err(codec->dev, "%s: Failed to set minmax "
+				"constraint: %d\n", __func__, ret);
+			return ret;
+		}
+	}
+
+	return 0;
+}
+
+static const struct snd_soc_dai_ops pcm5102a_dai_ops = {
+	.startup = pcm5102a_dai_startup,
+};
+
 static struct snd_soc_dai_driver pcm5102a_dai = {
 	.name = "pcm5102a-hifi",
 	.playback = {
 		.channels_min = 2,
 		.channels_max = 2,
-		.rates = SNDRV_PCM_RATE_8000_192000,
+		.rates = SNDRV_PCM_RATE_CONTINUOUS,
```

Comment:
You should use KNOT here instead of CONTINUOUS and explicitly list the supported rates via constraints. According to the datasheet the pcm5102a doesn't seem to support arbitrary rates (like 12345kHz)